### ISSUE TITLE
Feature Idea: Checkbox content type multiple checkboxes

### ISSUE THREAD

**Issue by ZevSwe** [mattermost/focalboard#288]
**Summary:**
Today when I add a checkbox it seems like I can just add one item. To add multiple check boxes one thus needs to add multiple content boxes. This might be a bug on my system, or it's just not fully implemented.

It would be nice with a multilevel multiple checkbox style content box. 

Example

Packing list:

- [ ] Clothes
   - [ ] Jacket
   - [ ] T shirts
   - [ ] underwear
   - [ ] Hoodie
- [ ] Toiletries  
   - [ ] Toothbrush
   - [ ] Toothpaste
   - [ ] Plackers
- [ ] Electronics
   - [ ] Computer
   - [ ] phone charger
   - [ ] computer charger
   - [ ] camera

**How important this is to me and why:**

Importance: Medium
Really useful feature

Use cases:
1. Make quick checklists in cards sorted into neat groups

**Comment by chenilim** [comment]
Thanks @ZevSwe! One way to implement this might be to add an "indent" action to each content block.

**Comment by ZevSwe** [comment]
Sure, but if I create a list of 20 items, I would not want to create 20 content blocks manually, that would be cumbersome. 

One way around that could be to make enter in a content block of type checklist create a new content block of the same type and indent.

**Comment by chenilim** [comment]
Yup, agreed. We need to make the general editing of content blocks easier, e.g.
* Pressing Enter when editing a checkbox label inserts a new checkbox below, and starts editing it
* Hitting backspace when editing an empty checkbox label deletes it and starts editing the checkbox immediately above it, if there is one.
* (Future) Pressing tab when editing a checkbox indents it

Rules like these would make editing items feel seamless.

**Comment by chmac** [comment]
If checkboxes inside the markdown body were clickable and considered "subtasks", this might solve this issue. I mentioned the idea on #252 but cross posting it here also. It's a bit wonky from a UX perspective that in this screenshot only 2 of the 5 "checkboxes" are actually clickable:

<img width="258" alt="Screenshot 2021-08-18 at 12 45 35" src="https://user-images.githubusercontent.com/690997/129885357-e841bcfd-5c0a-451a-b6cf-9b2ac3889f19.png">

**Comment by chenilim** [comment]
Thanks @chmac! Sorry, I missed your earlier comment in the other issue. Agree that it would be great if the Markdown editor supported clicking the checkboxes to toggle their state. Note that currently, you can "check" them in markdown with this format:
```
- [X] Checked item
- [ ] Unchecked item
```
I did a quick search, and this might be doable by [replacing the click handlers in CodeMirror](https://stackoverflow.com/questions/62684743/sync-checkbox-click-in-html-to-content-of-markdown-editor) (which is contained in the SimpleMDE control). Open to anyone who wants to take a closer look.
